source/common/init/target_impl.cc

```diff
@@ -27,19 +27,27 @@ TargetImpl::TargetImpl(absl::string_view name, InitializeFn fn)
       fn_(std::make_shared<InternalInitalizeFn>([this, fn](WatcherHandlePtr watcher_handle) {
         watcher_handle_ = std::move(watcher_handle);
         fn();
+        if (is_ready_) {
+          ready();
+        }
       })) {}
 
 TargetImpl::~TargetImpl() { ENVOY_LOG(debug, "{} destroyed", name_); }
 
 absl::string_view TargetImpl::name() const { return name_; }
 
 TargetHandlePtr TargetImpl::createHandle(absl::string_view handle_name) const {
+  if (is_ready_) {
```

Comment:
See note above in `ManagerImpl`. I think `createHandle` should always return a valid handle.